<launch>
    <!-- x方向滤波，限制在-5~5米 -->
    <node pkg="nodelet" type="nodelet" name="passthrough_filter_x" args="standalone pcl/PassThrough">
        <remap from="~input" to="/livox/lidar"/>
        <remap from="~output" to="/livox/lidar_x"/>
        <param name="filter_field_name" value="x"/>
        <param name="filter_limit_min" value="-5.0"/>
        <param name="filter_limit_max" value="5.0"/>
    </node>
    <!-- y方向滤波，限制在-5~5米 -->
    <node pkg="nodelet" type="nodelet" name="passthrough_filter_y" args="standalone pcl/PassThrough">
        <remap from="~input" to="/livox/lidar_x"/>
        <remap from="~output" to="/livox/lidar_xy"/>
        <param name="filter_field_name" value="y"/>
        <param name="filter_limit_min" value="-5.0"/>
        <param name="filter_limit_max" value="5.0"/>
    </node>

    <!-- octomap_server，实时生成八叉树和2D地图，输入为滤波后的点云 -->
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server" output="screen">
        <param name="frame_id" value="livox_frame"/>
        <param name="resolution" value="0.1"/>
        <param name="min_height" value="-1.0"/>
        <param name="max_height" value="2.0"/>
        <param name="latch" value="true"/>
        <param name="filter_ground" value="false"/>
        <param name="filter_speckles" value="false"/>
        <remap from="cloud_in" to="/livox/lidar_xy"/>
    </node>

    <!-- 可选：自动保存2D地图（如需静态地图） -->
    <!--<node pkg="map_server" type="map_saver" name="map_saver" output="screen">
        <param name="map_topic" value="/projected_map"/>
        <param name="save_dir" value="$(env HOME)/catkin_ws/maps"/>
    </node>-->
</launch>